src/lib/pubSub/index.ts

```diff
@@ -42,7 +46,11 @@ const pubSubFactory = <
     publish: (topic, info) => {
       if (topics[topic]) {
         topics[topic].forEach((fn) => {
-          setTimeout(() => fn(info !== undefined ? info : {}), 0);
+          if (opts?.publishSynchronous) {
```

Comment:
What's the purpose of synchronous?

Reply:
When publishing events, you can choose between synchronous and asynchronous execution.
Here are the differences illustrated with examples:

## Synchronous Execution:

```js
fns.forEach(fn => fn());
```
- Blocking: Each function runs to completion before the next one starts, potentially causing delays if any function takes time.
- Predictable Order: Functions execute in the exact order they are called.

## Asynchronous Execution:
```js
fns.forEach(fn => setTimeout(() => fn(), 0));
```
- Non-blocking: Functions are scheduled to run after the current execution context, keeping the main thread free.
- Less Predictable Order: Functions might not execute in the strict order they are scheduled due to other asynchronous tasks.